endpoints/openrtb2/video_auction.go

```diff
@@ -0,0 +1,495 @@
+				}
+				impsArray[impInd].Video.MaxDuration = int64(videoDur[durationIndex])
+				impsArray[impInd].Video.MinDuration = int64(videoDur[durationIndex])
+				//fmt.Println(podIndex, "  ", impInd, "duration ", videoDur[durationIndex])
+			} else {
+				impsArray[impInd].Video.MaxDuration = int64(maxDuration)
+			}
+
+			impsArray[impInd].ID = fmt.Sprintf("%d_%d", pod.PodId, impInd)
+		}
+		finalImpsArray = append(finalImpsArray, impsArray...)
+
+	}
+	return finalImpsArray, nil
+}
+
+func createImpressionTemplate(imp openrtb.Imp, video openrtb_ext.SimplifiedVideo) openrtb.Imp {
+	imp.Video = &openrtb.Video{}
+	imp.Video.W = video.W
+	imp.Video.H = video.H
+	imp.Video.Protocols = video.Protocols
+	imp.Video.MIMEs = video.Mimes
+	return imp
+}
+
+func loadStoredImp(storedImpId string) openrtb.Imp {
+	//temporary stub to test endpoint: placement id in place of pod config id
+	ext := fmt.Sprintf(`{"appnexus":{"placementId": %s}}`, storedImpId) //12971250
+
+	return openrtb.Imp{
+		ID:  "stored_imp_id",
+		Ext: []byte(ext)}
+
+}
+
+func minMax(array []int) (int, int) {
+	var max = array[0]
+	var min = array[0]
+	for _, value := range array {
+		if max < value {
+			max = value
+		}
+		if min > value {
+			min = value
+		}
+	}
+	return min, max
+}
+
+func buildVideoResponse(bidresponse *openrtb.BidResponse) (*openrtb_ext.BidResponseVideo, error) { //should be video response
+
+	adPods := make([]*openrtb_ext.AdPod, 0)
+	for _, seatBid := range bidresponse.SeatBid {
+		for _, bid := range seatBid.Bid {
+
+			var tempRespBidExt openrtb_ext.ExtBid
+			if err := json.Unmarshal(bid.Ext, &tempRespBidExt); err != nil {
+				return nil, err
+			}
+
+			impId := bid.ImpID
+			podNum := strings.Split(impId, "_")[0]
+			podInd, _ := strconv.ParseInt(podNum, 0, 64)
+
+			videoTargeting := openrtb_ext.VideoTargeting{
+				Hb_pb:         tempRespBidExt.Prebid.Targeting["hb_pb"],
+				Hb_pb_cat_dur: tempRespBidExt.Prebid.Targeting["hb_pb_cat_dur"],
+				Hb_cache_id:   tempRespBidExt.Prebid.Targeting["hb_cache_id"],
+			}
+
+			adPod := findAdPod(podInd, adPods)
+			if adPod == nil {
+				adPod = &openrtb_ext.AdPod{
+					PodId:     podInd,
+					Targeting: make([]openrtb_ext.VideoTargeting, 0, 0),
+				}
+				adPods = append(adPods, adPod)
+			}
+			adPod.Targeting = append(adPod.Targeting, videoTargeting)
+
+		}
+	}
+	videoResponse := openrtb_ext.BidResponseVideo{}
+	videoResponse.AdPods = adPods
+
+	return &videoResponse, nil
+}
+
+func findAdPod(podInd int64, pods []*openrtb_ext.AdPod) *openrtb_ext.AdPod {
+	for _, pod := range pods {
+		if pod.PodId == podInd {
+			return pod
+		}
+	}
+	return nil
+}
+
+func loadStoredVideoRequest(storedRequestId string) ([]byte, error) {
+	jsonString := []byte(`{"accountid": "11223344", "site": {"page": "mygame.foo.com"}}`)
+	return jsonString, nil
+}
+
+func getVideoStoredRequestId(request []byte) (string, error) {
+	req := openrtb_ext.StoredRequestId{}
+
+	if err := json.Unmarshal(request, &req); err != nil {
```

Comment:
jsonparser is probably a better fit here than parsing the entire request.

Reply:
Done